tests_new/integration_tests/aws_clients/s3.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from botocore.exceptions import ClientError
+
+log = logging.getLogger(__name__)
+
+
+class S3Client:
+    def __init__(self, session, region):
+        self._client = session.client('s3', region_name=region)
+        self._resource = session.resource('s3', region_name=region)
+        self._region = region
+
+    def delete_bucket(self, bucket_name):
```

Comment:
you need to account for access points as well (if bucket have them it will fail)

Reply:
In a happy path world, we should not run into that situation because a dataset can only be deleted if all shares are revoked and if shares are revoked the access points are deleted. However, if the tests for revoking access points fail, then we would face this situation when tearing down resources. I am going to check if it is  easy to implement :)

Reply:
Tested! Now it is ready to be merged. Do you want to have a final look @petrkalos ?